File: public/docs/_examples/hierarchical-dependency-injection/dart/lib/hero_card_component.dart
// #docregion
import 'package:angular2/core.dart';

import 'hero.dart';

@Component(
    selector: 'hero-card',
    template: '''
  <div>
    <span>Name:</span>
    <span>{{hero.name}}</span>
  </div>
  ''')
class HeroCardComponent {
  @Input() Hero hero;
}
// #docregion

File: public/docs/_examples/hierarchical-dependency-injection/ts/app/hero-card.component.ts
// #docregion
import { Component, EventEmitter, Input, Output } from '@angular/core';

import { Hero }             from './hero';
import { HeroStateService } from './hero-state.service';

@Component({
  moduleId: module.id,
  selector: 'hero-card',
  templateUrl: './hero-card.component.html',
  styleUrls: [ './hero-card.component.css' ],
  // #docregion providers
  providers: [ HeroStateService ]
  // #enddocregion providers
})
export class HeroCardComponent {

  @Output() close = new EventEmitter<void>();

  @Input()
  get hero() {
    return this.heroStateService.hero;
  }
  set hero (hero: Hero) {
    this.heroStateService.hero = hero;
  }

  constructor(private heroStateService: HeroStateService ) { }

  onCanceled()  {
    this.heroStateService.restoreHero();
  };

  onClose()  {
    this.close.emit();
  };

  onSaved() {
    this.heroStateService.saveHero();
  }
}
